yt_dlp/extractor/orf.py

```diff
@@ -604,10 +608,33 @@ def _extract_video(self, video_id, display_id):
                 formats.extend(fmts)
                 self._merge_subtitles(subs, target=subtitles)
 
+        for subtitle_type in ['vtt']:  # not working formats 'xml', 'srt', 'sami', 'ttml', 'stl'
```

Comment:
I cannot replicate this. For me, all subtitle formats on [this](https://on.orf.at/video/14210000/school-of-champions-48) video work. Do you have examples where they don't work?

Reply:
`stl` is mostly null in the api.
if I try `ttml`
```
~/p/yt-dlp on_orf_at_more_parsing• ❱ python3 yt_dlp/__main__.py https://on.orf.at/video/14226999 --sub-format ttml --embed-subs
[orf:on] Extracting URL: https://on.orf.at/video/14226999
[orf:on] 14226999: Downloading webpage
[orf:on] 14226999: Downloading JSON metadata
[orf:on] 14226999: Downloading m3u8 information
[orf:on] 14226999: Downloading m3u8 information
[orf:on] 14226999: Downloading MPD manifest
[orf:on] 14226999: Downloading MPD manifest
[info] 14226999: Downloading subtitles: de
[info] 14226999: Downloading 1 format(s): dash-p0va0br2998536-1+dash-p0aa0br192000-1
Deleting existing file ZIB 9_00 vom 20.05.2024 [14226999].de.ttml
[info] Writing video subtitles to: ZIB 9_00 vom 20.05.2024 [14226999].de.ttml
[download] Destination: ZIB 9_00 vom 20.05.2024 [14226999].de.ttml
[download] 100% of    7.99KiB in 00:00:00 at 87.87KiB/s
[download] ZIB 9_00 vom 20.05.2024 [14226999].mp4 has already been downloaded
[EmbedSubtitle] Embedding subtitles in "ZIB 9_00 vom 20.05.2024 [14226999].mp4"
ERROR: Postprocessing: file:ZIB 9_00 vom 20.05.2024 [14226999].de.ttml: Invalid data found when processing input
```

`python3 yt_dlp/__main__.py https://on.orf.at/video/14226999 --sub-format ttml --write-subs `
does work.
Same with the other formats.

I'm not so into file formats, but `vtt` was the only file format, that I was able to `--embed-sub` was working.

Can add the other one but then `--embed-sub` fails without `--sub-format vtt`.

Reply:
This is probably ffmpeg not being able to parse these subtitle files.
I would still like to extract all formats. It is possible to prioritize formats by the order which they are added.

I was thinking something like
```python
for sub_url in traverse_obj( api_json, ('_embedded', 'subtitle', lambda k, _: k.endswith('_url'), {url_or_none})):
    self._merge_subtitles({'de': [{'url': sub_url}]}, target=subtitles)
```
But that would no longer prioritize VTT. Maybe a separate step can be used to sort the subtitles?

Reply:
Alternatively, and even though I find it a bit ugly we could do:
```python
for sub_url in traverse_obj(api_json, (
        '_embedded', 'subtitle',
        ('xml_url', 'sami_url', 'stl_url', 'ttml_url', 'srt_url', 'vtt_url'), {url_or_none})):
    self._merge_subtitles({'de': [{'url': sub_url}]}, target=subtitles)
````

Reply:
I will try out some stuff and give then feedback

Reply:
this would work, but it has some assumptions:
```python
        for sub_url in traverse_obj(api_json, (
                '_embedded', 'subtitle',
                ..., {url_or_none})):
            subtitle_type = sub_url.split('.')[-1]
            self._merge_subtitles({'de': [{'url': sub_url, 'ext': subtitle_type}]}, target=subtitles)
        subtitles['de'].sort(
            # sort that vtt is default so `--embed-subs` is working out of the box
            key=lambda sub: 1 if sub['ext'] == 'vtt' else 0
        )
```

`traverse_obj` works because there is no other url in this path
workaround for ffmpeg: sort the subtitles after that so that `vtt` is in the front

will think a bit about this

Reply:
Yeah, thinking about this, I'd say sorting afterwards probably just overcomplicates things.
I would go with my second proposed solution of hardcoding all the keys and having vtt as the last one https://github.com/yt-dlp/yt-dlp/pull/9677#discussion_r1606691697

If you can push that, I'll approve this PR. Maybe the second reviewer later will have a better idea.

Reply:
I pushed a solution.
have all formats, and have `vtt` as default.
It's a bit ugly because of the sort after it.
except from that it should work

Reply:
> Alternatively, and even though I find it a bit ugly, we could do:
> 
> ```python
> for sub_url in traverse_obj(api_json, (
>         '_embedded', 'subtitle',
>         ('xml_url', 'sami_url', 'stl_url', 'ttml_url', 'srt_url', 'vtt_url'), {url_or_none})):
>     self._merge_subtitles({'de': [{'url': sub_url}]}, target=subtitles)
> ```

I didn't recognice, that the  travers_obj keep the sorting.
I like that solution

Reply:
for sub_url in traverse_obj(api_json, (
        '_embedded', 'subtitle',
        ('xml_url', 'sami_url', 'stl_url', 'ttml_url', 'srt_url', 'vtt_url'), {url_or_none})):
    self._merge_subtitles({'de': [{'url': sub_url}]}, target=subtitles)

Reply:
for sub_url in traverse_obj(api_json, (
        '_embedded', 'subtitle',
        ('xml_url', 'sami_url', 'stl_url', 'ttml_url', 'srt_url', 'vtt_url'), {url_or_none})):
    self._merge_subtitles({'de': [{'url': sub_url}]}, target=subtitles)